nef-plugin/SourceEditorCommand.swift

```diff
@@ -2,79 +2,112 @@
+        guard Reachability.isConnected else { return .failure(.internetConnection) }
+        guard let selection = editor.selection else { return .failure(.selection) }
+        
+        let appscheme = AppScheme(command: .exportSnippet(selection: selection),
+                                  estimatedDuration: .now() + .seconds(5))
         
-        AppScheme(action: .carbon(selection: selection)).run()
-        terminate(deadline: .now() + .seconds(5), completion)
+        return .success(appscheme)
     }
     
-    private func clipboardCarbon(editor: Editor, completion: @escaping (Error?) -> Void) {
-        guard Reachability.isConnected else { completion(EditorError.internetConnection); return }
-        guard let selection = editor.selection else { completion(EditorError.selection); return }
+    private func exportSnippetToClipboard(editor: Editor) -> Result<AppScheme, EditorError> {
+        guard Reachability.isConnected else { return .failure(.internetConnection) }
+        guard let selection = editor.selection else { return .failure(.selection) }
+        
+        let appscheme = AppScheme(command: .exportSnippetToClipboard(selection: selection),
+                                  estimatedDuration: .now() + .seconds(5))
         
-        AppScheme(action: .clipboardCarbon(selection: selection)).run()
-        terminate(deadline: .now() + .seconds(5), completion)
+        return .success(appscheme)
     }
     
-    private func markdownPage(editor: Editor, completion: @escaping (Error?) -> Void) {
-        guard editor.contentUTI == .playground || editor.contentUTI == .playgroundPage else { completion(EditorError.noPlayground); return }
+    private func markdownPage(editor: Editor) -> Result<AppScheme, EditorError> {
+        guard editor.contentUTI == .playground || editor.contentUTI == .playgroundPage else {
+            return .failure(.playgroundNotFound)
+        }
         
-        AppScheme(action: .markdownPage(playground: editor.code)).run()
-        terminate(deadline: .now(), completion)
+        let appscheme = AppScheme(command: .markdownPage(playground: editor.code))
+        return .success(appscheme)
     }
     
-    private func playgroundBook(editor: Editor, completion: @escaping (Error?) -> Void) {
-        guard editor.contentUTI == .package else { completion(EditorError.noPackage); return }
+    private func playgroundBook(editor: Editor) -> Result<AppScheme, EditorError> {
+        guard editor.contentUTI == .package else {
+            return .failure(.packageNotFound)
+        }
         
-        AppScheme(action: .playgroundBook(package: editor.code)).run()
-        terminate(deadline: .now(), completion)
+        let appscheme = AppScheme(command: .playgroundBook(package: editor.code))
+        return .success(appscheme)
     }
+}
+
+// MARK: - Terminate <helpers>
+extension Result where Success == AppScheme, Failure == EditorError {
```

Comment:
These methods should not be an extension over Result. If these functions only work on specific types for the success and failure cases, they should be functions receiving a result as input. Extensions should be as generic as possible.

Reply:
Using `fold` together terminate + terminateError methods.